modules/xhbBidAdapter.js

```diff
@@ -0,0 +1,457 @@
+   *
+   * @param {*} serverResponse A successful response from the server.
+   * @return {Bid[]} An array of bids which were nested inside the server.
+   */
+  interpretResponse: function(serverResponse, {bidderRequest}) {
+    serverResponse = serverResponse.body;
+    const bids = [];
+    if (!serverResponse || serverResponse.error) {
+      let errorMessage = `in response for ${bidderRequest.bidderCode} adapter`;
+      if (serverResponse && serverResponse.error) { errorMessage += `: ${serverResponse.error}`; }
+      utils.logError(errorMessage);
+      return bids;
+    }
+
+    if (serverResponse.tags) {
+      serverResponse.tags.forEach(serverBid => {
+        const rtbBid = getRtbBid(serverBid);
+        if (rtbBid) {
+          if (rtbBid.cpm !== 0 && includes(this.supportedMediaTypes, rtbBid.ad_type)) {
+            const bid = newBid(serverBid, rtbBid, bidderRequest);
+            bid.mediaType = parseMediaType(rtbBid);
+            bids.push(bid);
+          }
+        }
+      });
+    }
+    return bids;
+  },
+
+  getUserSyncs: function(syncOptions) {
+    if (syncOptions.iframeEnabled) {
+      return [{
+        type: 'iframe',
+        url: '//acdn.adnxs.com/ib/static/usersync/v3/async_usersync.html'
+      }];
+    }
+  }
+};
+
+function newRenderer(adUnitCode, rtbBid, rendererOptions = {}) {
+  const renderer = Renderer.install({
+    id: rtbBid.renderer_id,
+    url: rtbBid.renderer_url,
+    config: rendererOptions,
+    loaded: false,
+  });
+
+  try {
+    renderer.setRender(outstreamRender);
+  } catch (err) {
+    utils.logWarn('Prebid Error calling setRender on renderer', err);
+  }
+
+  renderer.setEventHandlers({
+    impression: () => utils.logMessage('xhb outstream video impression event'),
+    loaded: () => utils.logMessage('xhb outstream video loaded event'),
+    ended: () => {
+      utils.logMessage('xhb outstream renderer video event');
+      document.querySelector(`#${adUnitCode}`).style.display = 'none';
+    }
+  });
+  return renderer;
+}
+
+/* Turn keywords parameter into ut-compatible format */
+function getKeywords(keywords) {
+  let arrs = [];
+
+  utils._each(keywords, (v, k) => {
+    if (utils.isArray(v)) {
+      let values = [];
+      utils._each(v, (val) => {
+        val = utils.getValueString('keywords.' + k, val);
+        if (val) { values.push(val); }
+      });
+      v = values;
+    } else {
+      v = utils.getValueString('keywords.' + k, v);
+      if (utils.isStr(v)) {
+        v = [v];
+      } else {
+        return;
+      } // unsuported types - don't send a key
+    }
+    arrs.push({key: k, value: v});
+  });
+
+  return arrs;
+}
+
+/**
+ * Unpack the Server's Bid into a Prebid-compatible one.
+ * @param serverBid
+ * @param rtbBid
+ * @param bidderRequest
+ * @return Bid
+ */
+function newBid(serverBid, rtbBid, bidderRequest) {
+  const bid = {
+    requestId: serverBid.uuid,
+    cpm: 0.00,
+    creativeId: rtbBid.creative_id,
+    dealId: 99999999,
```

Comment:
The `dealId` field is an optional field and doesn't have to be explicitly set.  If you're not planning on using it, it could be removed instead of hard-coding to a set value.  If you are using that set value for a particular use-case, please let me know.

Reply:
This is necessary for our setup.